app/models/account_authorization_config/clever.rb

```diff
@@ -64,7 +65,17 @@ def provider_attributes(token)
   protected
 
   def me(token)
-    token.options[:me] ||= token.get("/me").parsed['data']
+    token.options[:me] ||= begin
+                             raw_data = token.get("/v2.1/me").parsed
+                             data = raw_data["data"].dup
```

Comment:
I feel like we may be able to clean up this method a little bit...

Some questions that I have looking at this: 
1. Why are creating a variable `data` by calling `.dup` on the `raw_data` variable?
2. We are making 2 `token.get` calls for different endpoints... can you explain what each is doing so that we can determine if there is a more efficient way to get the pieces of data we need?
3. In the second `token.get` call, you have `#{raw_data["type"]}s/#{data["id"]}"`... can you break down what this interpolation should look like in practice? Particularly, the `s/...` raises some concerns. 
4. I like the use of `dig` here, but what happens if any of the `dig`s return `nil`?

Reply:
+1

in particular line 71... how might we make this more readable?

Reply:
1. Why are creating a variable data by calling .dup on the raw_data variable?
- I wanted to preserve the original response by creating a shallow copy. Not entirely sure what else uses that response data so I wanted to basically just use the new "data" variable as a read-only copy of the actual response "raw_data".

2. We are making 2 token.get calls for different endpoints... can you explain what each is doing so that we can determine if there is a more efficient way to get the pieces of data we need?
- Clever changed their endpoints to return different amounts of data. The endpoint /v2.1/me only returns certain fields from the user object. The documentation doesn't state exactly what it returns, but it does return the type of user:
`https://dev.clever.com/v2.1/reference/me no longer returns the full user object for Secure Sync users. All integrations should subsequently call /<user type>/<id> using the Data API (and corresponding district-app token) to access the user object.`
I then take the type from the raw_data response to formulate a new api call to abide by their new API route. Similar to:
`/v2.1/students/4fee004cca2e43cf27000007/`


3. In the second token.get call, you have #{raw_data["type"]}s/#{data["id"]}"... can you break down what this interpolation should look like in practice? Particularly, the s/... raises some concerns.
Similar to what I said above, it is just how we return the "students" data for a particular student. For some reason, the type returns as "student" so I added the "s" to complete that:

`/v2.1/students/4fee004cca2e43cf27000007/`

4. I like the use of dig here, but what happens if any of the digs return nil?
- Previously, I was accessing data from the response incorrectly, so it was returning nil. In that instance, it returned an empty string. 

5. in particular line 71... how might we make this more readable?
- I am not entirely sure how else to do it. If there is a refactor I can do to make it more readable I'm happy to make the change.

Thank you @desertviper @CJAdeszko, hopefully this answers your questions.

Reply:
Refactored to hopefully improve readability.
@desertviper

Reply:
👍